service/dap/server.go

```diff
@@ -451,6 +451,44 @@ func (s *Server) onContinueRequest(request *dap.ContinueRequest) {
 	s.doContinue()
 }
 
+func (s *Server) onThreadsRequest(request *dap.ThreadsRequest) {
+	if s.debugger == nil {
+		s.sendErrorResponse(request.Request, UnableToDisplayThreads, "Unable to display threads", "debugger is nil")
+		return
+	}
+	gs, _, err := s.debugger.Goroutines(0, 0)
+	if err != nil {
+		switch err.(type) {
+		case *proc.ErrProcessExited:
+			// If the program exits very quickly, the initial threads request will complete after it has exited.
+			// A TerminatedEvent has already been sent. Ignore the err returned in this case.
+			s.send(&dap.ThreadsResponse{Response: *newResponse(request.Request)})
+		default:
+			s.sendErrorResponse(request.Request, UnableToDisplayThreads, "Unable to display threads", err.Error())
+		}
+		return
+	}
+
+	threads := make([]dap.Thread, len(gs))
+	if len(threads) == 0 {
+		threads = []dap.Thread{{Id: 1, Name: "Dummy"}}
+	} else {
+		for i, g := range gs {
+			threads[i].Id = g.ID
+			if g.UserCurrentLoc.Function != nil {
```

Comment:
if fn := g.UserCurrentLoc.Function; fn != nil {
   threads[i].Name = fn.Name()
} else {
   threads[i].Name = fmt.Sprintf(..)
}

Reply:
I wanted to shorten the line :-) If you desire to make it clear that the if/else share the same g.UserCurrentLoc, 
```
if loc := g.UserCurrentLoc; loc.Function != nil {
   threads[i].Name = loc.Function.Name()
} else {
   threads[i].Name = fmt.Sprintf("%s@%d", loc.File, loc.Line)
}
```